frontend/src/app/main/ui/workspace/sidebar/assets.cljs

```diff
@@ -270,57 +271,209 @@
 
 ;; ---- Components box ----
 
+(def components-to-group (atom []))
+
+(defn- create-assets-group
+  [rename group-name]
+  (st/emit! (dwu/start-undo-transaction))
+  (apply st/emit!
+         (->> @components-to-group
+              (map #(rename
+                     (:id %)
+                     (add-group % group-name)))))
+  (st/emit! (dwu/commit-undo-transaction)))
+
+
+(defn- on-drop-asset
+  [event asset dragging? selected-assets selected-assets-full selected-assets-paths rename]
+  (let [create-typed-assets-group (partial create-assets-group rename)]
+    (when (not (dnd/from-child? event))
+      (reset! dragging? false)
+      (when
+       (and (not (contains? selected-assets (:id asset)))
+            (every? #(= % (:path asset)) selected-assets-paths))
+        (reset! components-to-group (conj selected-assets-full asset))
```

Comment:
I think that the components to group could be simply passed to the modal as another parameter, instead of using a global atom

Reply:
Good catch, fixed